unpack

```diff
@@ -2,70 +2,38 @@
 # unpack - unarchive archive files
 # this file in public domain
 
-usage()
+die()
 {
-	echo "usage: unpack file [destdir]" >&2
+	printf "%s\n" "$*" >&2
 	exit 1
 }
 
+crun()
+{
+	command -v "$1" 2>&1 >/dev/null ||
+		die "unpack: utility $1 is not installed"
+
+	"$@" || die "failed to extract"
+}
+
 case "$#" in
-(1) set -- "$1" "$PWD" ;;
-(2)                    ;;
-(*) usage              ;;
+	(1) set -- "$1" "$PWD" ;;
+	(2) ;;
+	(*) die "usage: unpack file [destdir]" ;;
 esac
 
-if [ ! -f "$1" ]
-then
-	printf "unpack: %s: not a regular file\n" "$1"
-	exit 1
-fi
-
-if [ ! -d "$2" ]
-then
-	printf "unpack: %s: not a regular file\n" "$2"
-	exit 1
-fi
+! [ -e "$1" ] && die "unpack: $1: not a valid file"
+! [ -d "$2" ] && die "unpack: $2: not a valid directory"
 
-case "$(file -ib -- "$1" | cut -d';' -f1 | cut -d'/' -f2)" in
-gzip|x-gzip)
-	tar xzvf "$1" -C "$2" >/dev/null || {
-	echo "Package can not be extracted, try to install 'tar'" &&
-	exit 1 ; }
-	;;
-bzip2|x-bzip2)
-	tar xjvf "$1" -C "$2" >/dev/null || {
-	echo "Package can not be extracted, try to install 'tar'" &&
-	exit 1 ; }
-	;;
-tar|x-tar)
-	tar xvf "$1" -C "$2" >/dev/null || {
-	echo "Package can not be extracted, try to install 'tar'" &&
-	exit 1 ; }
-	;;
-zip)
-	unzip -d "$2" -- "$1" >/dev/null || {
-	echo "Package can not be extracted, try to install 'unzip'" &&
-	exit 1 ; }
-	;;
-x-rar)
-	unrar x "$1" "$2" >/dev/null || {
-	echo "Package can not be extracted, try to install 'unrar'" &&
-	exit 1 ; }
-	;;
-x-7z-compressed)
-	7z x -o"$2" "$1" >/dev/null || {
-	echo "Package can not be extracted, try to install '7-zip'" &&
-	exit 1 ; }
-	;;
-x-xz)
-	xzcat "$1" | tar xvf - -C "$2" >/dev/null || {
-	echo "Package can not be extracted, try to install 'xz'" &&
-	exit 1 ; }
-	;;
-*)
-	printf "unpack: %s: not an archive file\n" "$1"
-	exit 1
-	;;
+case "${1##*.}" in
+tgz|gz)  crun tar xzf "$1" -C "$2" ;;
```

Comment:
This is neat. unlike above I think this change made the code look nicer and less crowded with the `crun()` function usage.